### PR TITLE
notification on destroy oauth apps

### LINKED ISSUE #15016: When an oauth_app is deleted, we will just notify their users through the notification center.
Current suggested copy: The app #{app_name} has signed off. You may find more information in their %a{ href: #{website_url}, title: 'website' }.

### PR DESCRIPTION
**Pull request by alrocar**
Closes #15016

Merge after https://github.com/CartoDB/cartodb/pull/15023

### CODE REVIEW

File: app/models/carto/oauth_app.rb [hunk 1]
```diff
@@ -46,6 +49,26 @@ def regenerate_client_secret!
...
+    end
+
+    def send_destroy_notification
+      return if @user_ids.empty?
+      notification = Carto::Notification.create!(body: notification_body, icon: Notification::ICON_ALERT)
+      ::Resque.enqueue(::Resque::UserJobs::Notifications::Send, @user_ids, notification.id)
+    rescue StandardError => e
+      CartoDB::Logger.warning(
+        message: "Couldn't notify users about oauth_app '#{name}' deletion",
+        notification_id: notification&.id,
```
Comment (ethervoid) [inline app/models/carto/oauth_app.rb]: notification.id?

File: app/models/carto/oauth_app.rb [hunk 2]
```diff
@@ -46,6 +49,26 @@ def regenerate_client_secret!
 
     private
 
+    def collect_users
+      @user_ids = oauth_app_users.collect { |u| u.user.id }
+    end
+
+    def send_destroy_notification
```
Comment (ethervoid) [inline app/models/carto/oauth_app.rb]: Maybe something like `send_app_removal_notification`

File: app/models/carto/oauth_app.rb (hunk 1 above)
Comment (alrocar) [reply app/models/carto/oauth_app.rb]: It might not exist if the `Notification.create!` failed

**Comment by gonzaloriestra** [comment]
:heavy_check_mark: Acceptance OK

**Comment by gonzaloriestra** [comment]
retest this please

**Comment by gonzaloriestra** [comment]
restest this please

**Comment by gonzaloriestra** [comment]
retest this please

**Pull request merged**